.travis.yml

```diff
@@ -13,10 +11,33 @@ sudo: required
 before_install:
   - docker-compose -f .travis/docker-compose-travis.yml up -d
 
+install:
+    - pip install -r requirements/travis.txt
+
 script:
-  - docker exec analytics_api_testing /edx/app/analytics_api/analytics_api/.travis/run_tests.sh
+  - docker exec -t -e TRAVIS=1 analytics_api_testing bash -c "
+      cd /edx/app/analytics_api/analytics_api/ &&
+      make $TARGETS
+      "
 
-after_success:
-  - pip install -U codecov
-  - docker exec analytics_api_testing /edx/app/analytics_api/analytics_api/.travis/run_coverage.sh
-  - codecov
+matrix:
+  include:
+    - python: 2.7
+      env:
+        TESTNAME=quality
+        TARGETS="quality"
+    - python: 2.7
+      env:
+        TESTNAME=test-python
+        TARGETS="test"
+      after_success:
+        - docker exec analytics_api_testing /edx/app/analytics_api/analytics_api/.travis/run_coverage.sh
+        - codecov --disable pycov
```

Comment:
Out of curiosity, why did you have to add `--disable pycov`?

Reply:
I noticed that codecov collects reports and then it runs `coverage xml` but since that report was already done and is in build/coverage/coverage.xml I think that it is not necessary. 

https://github.com/codecov/codecov-python/blob/3a8b06b2c7de69bfa6a7fa8eb2517df558b39342/codecov/__init__.py#L678

On the other hand, I noticed that in the current builds, the `coverage xml` command that is made by codecov is failing because coverage is not installed. That would be solved if we add that package to the travis.in file but still I think that it is not necessary to run that command again.